library/Fission/User/Mutation.hs

```diff
@@ -100,9 +101,8 @@ hashPassword' password = do
     Nothing           -> Left Error.FailedDigest
     Just secretDigest -> Right <| decodeUtf8Lenient secretDigest
 
-destroy :: MonadDB m => UserId -> Transaction m ()
-destroy userId = delete <| from \user ->
-    where_ (user ^. UserId ==. val userId)
+destroy :: MonadDBMutation User m => UserId -> Transaction m ()
+destroy userId = deleteWhere (\user -> user ^. UserId ==. val userId)
```

Comment:
Also don't need the parens here:

```haskell
deleteWhere \user -> user ^. UserId ==. val userId
```

---

You know, if you wanted the simpler-style syntax, we could write something like:

```haskell
deleteWhere' matchOne = deleteWhere \model -> model ^. matchOne

-- Then:
destroy userId = deleteWhere (UserId ==. val userId)
```

Or even:

```haskell
deleteWhere'' andList = deleteWhere \model ->
  andList
    |> fmap (model ^.)
    |> foldr1 (&&.)

-- Use:

deleteWhere'' [UserId ==. val userId, UserEmail ==. val email. UserAge ==. val 42]

-- Same as:

deleteWhere \model ->
      model ^. UserId    ==. val UserId
  &&. model ^. UserEmail ==. val email
  &&. model ^. UserAge   ==. val 42
```